### PR TITLE
[Container app] Fix #28998: `az containerapp env workload-profile add/update`: Fix `NoneType` object is not iterable error when environment doesn't enable workload profile

### LINKED ISSUE #28998: Adding a workload profile to existing Container Apps Environment fails
### Describe the bug

Adding a workload profile to existing Container Apps Environment fails

### Related command

```
az containerapp env workload-profile add \
  --name -snip- \
  --resource-group -snip- \
  --workload-profile-name consumption \
  --workload-profile-type Consumption \
  --min-nodes 1 \
  --max-nodes 5
```

### Errors

```
The command failed with an unexpected error. Here is the traceback:
'NoneType' object is not iterable
Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 664, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 731, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 701, in _run_job
    result = cmd_copy(params)
             ^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 334, in __call__
    return self.handler(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/azure-cli/2.61.0/libexec/lib/python3.11/site-packages/azure/cli/command_modules/containerapp/custom.py", line 4932, in add_workload_profile
    workload_profiles_lower = [p["name"].lower() for p in workload_profiles]
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```

### Issue script & Debug output

```
unrecognized arguments:  --debug
```
😉 

### Expected behavior

A workload profile is created and added to the Container Apps Environment.

### Environment Summary

```
azure-cli                         2.61.0

core                              2.61.0
telemetry                          1.1.0

Extensions:
arcdata                           1.5.13
azure-devops                       1.0.1
containerapp                      0.3.52
resource-graph                     2.1.0

Dependencies:
msal                              1.28.0
azure-mgmt-resource               23.1.1

Python location '/usr/local/Cellar/azure-cli/2.61.0/libexec/bin/python'
Extensions directory '/Users/michaeleikelandtt/.azure/cliextensions'

Python (Darwin) 3.11.9 (main, Apr  2 2024, 08:25:04) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Greedygre**
**Related command**

Fix https://github.com/Azure/azure-cli/issues/28998

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Container app

File: src/azure-cli/azure/cli/command_modules/containerapp/custom.py [hunk 1]
```diff
@@ -5022,11 +5022,16 @@ def set_workload_profile(cmd, resource_group_name, env_name, workload_profile_na
...
 def add_workload_profile(cmd, resource_group_name, env_name, workload_profile_name, workload_profile_type=None, min_nodes=None, max_nodes=None):
+    r = None
     try:
         r = ManagedEnvironmentClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
     except CLIError as e:
         handle_raw_exception(e)
 
+    if r and safe_get(r, "properties", "workloadProfiles") is None:
+        raise ValidationError(f"Cannot add workload profile because the environment doesn't enable workload profile.\n"
+                              f"If you want to use Consumption and Dedicated environment, please create a new one.")
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/containerapp/custom.py:5033]: with --enable-workload-profile?

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/custom.py:5033]: Currently the default value of --enable-workload-profile has been set to `True`, no need to specify it when creating a new environment.

**Review by LaylaLiu-gmail — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/containerapp/custom.py [hunk 2]
```diff
@@ -5022,11 +5022,16 @@ def set_workload_profile(cmd, resource_group_name, env_name, workload_profile_na
...
 
 def add_workload_profile(cmd, resource_group_name, env_name, workload_profile_name, workload_profile_type=None, min_nodes=None, max_nodes=None):
+    r = None
     try:
         r = ManagedEnvironmentClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
     except CLIError as e:
         handle_raw_exception(e)
 
+    if r and safe_get(r, "properties", "workloadProfiles") is None:
+        raise ValidationError("Cannot add workload profile because the environment doesn't enable workload profile.\n"
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/containerapp/custom.py:5032]: create new one with `az containerapp enev create`

**Review by Juliehzl — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/containerapp/custom.py (hunk 2 above)
Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/custom.py:5032]: Fixed

**Review by Juliehzl — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**